rs/tests/system_tests.bzl

```diff
@@ -263,6 +274,50 @@ def system_test(
         flaky = flaky,
     )
 
+def system_test_nns(name, extra_head_nns_tags = ["system_test_nightly"], **kwargs):
+    """Declares a system-test that uses the mainnet NNS and a variant that use the HEAD NNS.
+
+    Declares two system-tests:
+
+    * One with the given name which uses the NNS from mainnet as specified by mainnet-canisters.bzl.
+    * One with the given name suffixed with "_head_nns" which uses the NNS from the HEAD of the repo.
+
+    The latter one is additionally tagged with "system_test_nightly" such that it only runs daily and not on PRs.
+    You can override the latter behaviour by specifying different `extra_head_nns_tags`.
+    If you set `extra_head_nns_tags` to `None` the head_nns variant will have the same tags as the default variant.
+
+    The idea being that for most system-tests which test the replica it's more realistic to test against the
+    mainnet NNS since that version would be active when the replica would be released.
+
+    However it's still useful to see if the HEAD replica works against the HEAD NNS which is why this macro
+    introduces the <name>_head_nns variant which only runs daily if not overriden.
+
+    Args:
+        name: the name of the system-tests.
+        extra_head_nns_tags: extra tags assigned to the head_nns variant or None to use the original tags.
+        **kwargs: the arguments of the system-tests.
+    """
+    runtime_deps = kwargs.pop("runtime_deps", [])
+    env = kwargs.pop("env", {})
+
+    system_test(
+        name,
+        env = env | MAINNET_NNS_CANISTER_ENV,
+        runtime_deps = runtime_deps + MAINNET_NNS_CANISTER_RUNTIME_DEPS,
+        **kwargs
+    )
+
+    original_tags = kwargs.pop("tags", [])
+    system_test(
+        name,
+        variant = "head_nns",
+        env = env | NNS_CANISTER_ENV,
+        runtime_deps = runtime_deps + NNS_CANISTER_RUNTIME_DEPS,
+        tags = [tag for tag in original_tags if tag not in extra_head_nns_tags] +
+               extra_head_nns_tags if extra_head_nns_tags != None else original_tags,
```

Comment:
`if extra_head_nns_tags != None` is unnecessary. If folks want to `original_tags` they can just set `extra_head_nns_tags` to `[]`.